packages/IMAPClient-Core.package/ICEmail.class/instance/retrieveBody.st

```diff
@@ -1,9 +1,7 @@
-endpoint interaction
+action
 retrieveBody
 	
-	self folder endpoint
-		select: self folder path;
-		fillBodyFor: self.
+	self folder endpoint retrieveBody: self.
```

Comment:
As the mail is responsible for retrieving its on body, there will have to be some kind of endpoint interaction here. One could think about having a reference on the endpoint in the ICMail object, so the communication does not have to run over the folder instance.

Reply:
Actually retrieving the body is the only action that really needs to be in the ICEmail, because it is called when the email is opened. If we move all other actions to the ICFolder, then the folder is reponsible for every action the ICFolderDialog offers and the Email is responsible for the ICEmailDialog. This does not resolve the communication over the folder though for the `retrieveBody` method.